src/libexpr/primops/fetchTree.cc

```diff
@@ -193,8 +193,52 @@ static void prim_fetchTree(EvalState & state, const PosIdx pos, Value * * args,
     fetchTree(state, pos, args, v, std::nullopt, FetchTreeParams { .allowNameArgument = false });
 }
 
-// FIXME: document
-static RegisterPrimOp primop_fetchTree("fetchTree", 1, prim_fetchTree);
+static RegisterPrimOp primop_fetchTree({
+    .name = "fetchTree",
+    .args = {"args"},
+    .doc = R"(
+      Fetch a source tree or a plain file using one of supported
+      backends. *args* can be an attribute set representation of
+      [flake reference](../command-ref/new-cli/nix3-flake.html#flake-references)
+      in which case the referenced tree is fetched. Otherwise, it can
+      be an URL.
```

Comment:
This is awkwardly phrased, since the referenced tree is also fetched in the URL case.